common/subtitle-coloring/subtitle-coloring.ts

```diff
@@ -0,0 +1,790 @@
+        const { text, index, ts } = options;
+        try {
+            if (!this.anki) throw new Error('Anki not initialized');
+            if (!ts.yomitanHealthy) throw new Error(`Yomitan not initialized for Track${ts.track + 1}`);
+            if (!ts.dt.dictionaryAnkiWordFields.length && !ts.dt.dictionaryAnkiSentenceFields.length) {
+                throw new Error('No Anki fields defined');
+            }
+
+            let richText: string = '';
+            let textHasError = false;
+            let textHasUncollected = false;
+            let rawTokens = ts.tokenizeCache.get(text);
+            if (!rawTokens) {
+                rawTokens = await this.yomitan.tokenize(
+                    text,
+                    ts.dt.dictionaryYomitanScanLength,
+                    ts.dt.dictionaryYomitanUrl
+                );
+                ts.tokenizeCache.set(text, rawTokens);
+                if (this.shouldCancelBuild) return;
+            }
+            for (const rawToken of rawTokens) {
+                const trimmedToken = rawToken.trim();
+
+                // Token is already cached or not a word
+                const cachedTokenStatus = ts.tokenStatusCache.get(trimmedToken);
+                if (this._tokenStatusValid(cachedTokenStatus)) {
+                    const richRes = this._applyTokenStyle({ rawToken, tokenStatus: cachedTokenStatus!, dt: ts.dt });
+                    if (richRes) richText += richRes;
+                    if (cachedTokenStatus === TokenStatus.UNCOLLECTED) textHasUncollected = true;
+                    else if (cachedTokenStatus === null) textHasError = true;
+                    continue;
+                }
+                if (!HAS_LETTER_REGEX.test(trimmedToken)) {
+                    const fullyKnownTokenStatus = getFullyKnownTokenStatus();
+                    const richRes = this._applyTokenStyle({ rawToken, tokenStatus: fullyKnownTokenStatus, dt: ts.dt });
+                    if (richRes) richText += richRes;
+                    ts.tokenStatusCache.set(trimmedToken, fullyKnownTokenStatus);
+                    continue;
+                }
+
+                let shouldCheckExactFormWordField = true;
+                if (ts.dt.dictionaryTokenMatchStrategy === TokenMatchStrategy.LEMMA_FORM_COLLECTED) {
+                    let tokenLemmas = ts.lemmatizeCache.get(trimmedToken);
+                    if (!tokenLemmas) {
+                        tokenLemmas = await this.yomitan.lemmatize(trimmedToken, ts.dt.dictionaryYomitanUrl);
+                        ts.lemmatizeCache.set(trimmedToken, tokenLemmas);
+                        if (this.shouldCancelBuild) return;
+                    }
+                    if (tokenLemmas.length) shouldCheckExactFormWordField = false;
+                }
+                let shouldCheckExactFormSentenceField = true;
+                if (ts.dt.dictionaryAnkiSentenceTokenMatchStrategy === TokenMatchStrategy.LEMMA_FORM_COLLECTED) {
```

Comment:
Would it make sense to remove the sentence-specific variant and just use `dictionaryTokenMatchStrategy`? I think it would be very rare for someone to want to configure a different strategy for word vs sentence fields.